refactor: echo 27: regenerate first release commit to review

luarmr opened the pull request:
This pr is how the release of the migation will look like,

Please front end guild review commit per commit, check here how to: https://humansignal.slack.com/archives/C03H1JNCZDH/p1712688816246619?thread_ts=1712687691.578269&cid=C03H1JNCZDH

Check the hook in local, and confirm this is our prefer method. 




### PR fulfills these requirements
- [ ] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend



### Describe the reason for change
_(link to issue, supportive screenshots etc.)_



#### What does this fix?
_(if this is a bug fix)_



#### What is the new behavior?
_(if this is a breaking or feature change)_



#### What is the current behavior?
_(if this is a breaking or feature change)_



#### What libraries were added/updated?
_(list all with version changes)_



#### Does this change affect performance?
_(if so describe the impacts positive or negative)_



#### Does this change affect security?
_(if so describe the impacts positive or negative)_



#### What alternative approaches were there?
_(briefly list any if applicable)_



#### What feature flags were used to cover this change?
_(briefly list any if applicable)_



### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

luarmr committed:
add hook

adding pre-commit hook

update hook to 1.7.0

luarmr committed:
excluding 2 random errors, follow up

luarmr committed:
adding biome to ci

luarmr commented:
@farioas I saw.you being ping due code owners.
This proposal is to plan the changes related to linting. Let's discuss next week about the best way to implement the new workflow. We can decide if it should be in a separate pull request. So for this specific PR no need to approve. It wont be the final.

juliosgarbi commented:
/git merge develop

juliosgarbi commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8741272691)
> **Error**: failed to execute "build" command

luarmr committed:
refactor: ECHO-27: introducting biome, removing remining eslint

luarmr committed:
first biome configuration excluding all unsafe and safe that are not corrected automaticly

luarmr committed:
Apply biome check corrections

luarmr committed:
Apply biome check corrections second passs

luarmr committed:
noAriaHiddenOnFocusable

luarmr committed:
noAriaUnsupportedElements

luarmr committed:
noHeaderScope

luarmr committed:
noDistractingElements

luarmr committed:
noInteractiveElementToNoninteractiveRole

luarmr committed:
noNoninteractiveElementToInteractiveRole

luarmr committed:
noPositiveTabindex

luarmr committed:
noNoninteractiveTabindex

luarmr committed:
noRedundantRoles

luarmr committed:
useAnchorContent

luarmr committed:
useValidAriaProps

luarmr committed:
useAriaActivedescendantWithTabindex

luarmr committed:
noExtraBooleanCast

luarmr committed:
useValidAriaRole

luarmr committed:
noUselessConstructor

luarmr committed:
noThisInStatic

luarmr committed:
noUselessFragments

luarmr committed:
noUselessLoneBlockStatements

luarmr committed:
noUselessTernary

luarmr committed:
useLiteralKeys

luarmr committed:
useRegexLiterals

luarmr committed:
useSimplifiedLogicExpression

luarmr committed:
noConstAssign

luarmr committed:
noInvalidNewBuiltin

luarmr committed:
noNewSymbol

luarmr committed:
noNonoctalDecimalEscape

luarmr committed:
noSwitchDeclarations

luarmr committed:
noStringCaseMismatch

luarmr committed:
noUnnecessaryContinue

luarmr committed:
noUnusedLabels

luarmr committed:
noUnusedVariables

luarmr committed:
noUnusedPrivateClassMembers

luarmr committed:
useIsNan

luarmr committed:
noVoidElementsWithChildren

luarmr committed:
noShoutyConstants

luarmr committed:
noUnusedTemplateLiteral

luarmr committed:
noVar

luarmr committed:
noUselessElse

luarmr committed:
useBlockStatements

luarmr committed:
useConsistentArrayType

luarmr committed:
useExponentiationOperator

luarmr committed:
useFragmentSyntax

luarmr committed:
useNumberNamespace

luarmr committed:
useSelfClosingElements

luarmr committed:
useNumericLiterals

luarmr committed:
useShorthandAssign

luarmr committed:
useShorthandArrayType

luarmr committed:
useSingleCaseStatement

luarmr committed:
useTemplate

luarmr committed:
useSingleVarDeclarator

luarmr committed:
noCommentText

luarmr committed:
noApproximativeNumericConstant

luarmr committed:
noConsoleLog

luarmr committed:
noDebugger

luarmr committed:
noDuplicateObjectKeys

luarmr committed:
noDoubleEquals

luarmr committed:
noGlobalIsFinite

luarmr committed:
noMisrefactoredShorthandAssign

luarmr committed:
noSparseArray

luarmr committed:
noUnsafeNegation

luarmr committed:
useIsArray

luarmr committed:
noSkippedTests

luarmr committed:
noFocusedTests

luarmr committed:
noConsole

luarmr committed:
useValidTypeof

luarmr committed:
useSortedClasses

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8747647487

luarmr commented:
Close in favor of:
https://github.com/HumanSignal/label-studio/pull/5759
the branch name is being problematic to generate the build

luarmr commented:
Close in favor of:
https://github.com/HumanSignal/label-studio/pull/5759
